### PR TITLE
[Doc] Add documentations on shared data cluster configurations

### PR DESCRIPTION
**Pull request by sduzh**
## What type of PR is this:

- [ ] BugFix
- [ ] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [x] Doc
- [ ] Tool

Does this PR entail a change in behavior?

- [ ] Yes, this PR will result in a change in behavior.
- [x] No, this PR will not result in a change in behavior.

If yes, please specify the type of change:

- [ ] Interface/UI changes: syntax, type conversion, expression evaluation, display information
- [ ] Parameter changes: default values, similar parameters but with different default values
- [ ] Policy changes: use new policy to replace old one, functionality automatically enabled
- [ ] Feature removed
- [ ] Miscellaneous: upgrade & downgrade compatibility, etc.

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:

- [x] I have checked the version labels which the pr will be auto-backported to the target branch
  - [x] 3.2
  - [ ] 3.1
  - [ ] 3.0
  - [ ] 2.5

### CODE REVIEW

**Review by wyb — DISMISSED** [review]

**Review by wyb — DISMISSED** [review]

**Review by kevincai — APPROVED** [review]

File: docs/zh/administration/Configuration.md [hunk 1]
```diff
@@ -607,6 +607,105 @@ ADMIN SET FRONTEND CONFIG ("key" = "value");
...
 
+#### 存算分离相关动态参数
+
+##### lake_compaction_score_selector_min_score
+
+- 含义：触发 Compaction 操作的 Compaction Score 阈值。当一个表分区的 Compaction Score 大于或等于该值时，系统会对该分区执行 Compaction 操作。
+- 默认值： 10.0
+- 引入版本：v3.2.0
+
+Compaction Score 代表了一个表分区是否值得进行 Compaction 的评分，您可以通过 [SHOW PARTITIONS](../sql-reference/sql-statements/data-manipulation/SHOW_PARTITIONS.md) 语句返回中的 `MaxCS` 一列的值来查看某个分区的 Compaction Score。Compaction Score 和分区中的文件数量有关系。文件数量过多将影响查询性能，因此系统后台会定期执行 Compaction 操作来合并小文件，减少文件数量。
```
Comment (evelynzhaojie) [inline docs/zh/administration/Configuration.md:618]: @sduzh MaxCS 这个返回列是存算分离表 show partitions 时新加的返回字段吗，如果 show partitions 存算分离相关返回字段有变化，请负责的 rd 提交文档 pr 改一下。

File: docs/zh/administration/Configuration.md [hunk 2]
```diff
@@ -607,6 +607,105 @@ ADMIN SET FRONTEND CONFIG ("key" = "value");
...
+
+- 含义：允许同时执行的 Compaction 任务数。
+- 默认值：-1
+- 引入版本：v3.2.0
+
+系统依据分区中 Tablet 数量来计算 Compaction 任务数。如果一个分区有 10 个 Tablet，那么对该分区作一次 Compaciton 就会创建 10 个 Compaction 任务。如果正在执行中的 Compaction 任务数超过该阈值，系统将不会创建新的 Compaction 任务。将该值设置为 `0` 表示禁止 Compaction，设置为 `-1` 表示系统依据自适应策略自动计算该值。
+
+##### lake_compaction_history_size
+
+- 含义：在 Leader FE 节点内存中保留多少条最近成功的 Compaction 任务历史记录。您可以通过 `SHOW PROC '/transactions'` 命令查看最近成功的 Compaction 任务记录。请注意，Compaction 历史记录是保存在 FE 进程内存中的，FE 进程重启后历史记录会丢失。
```
Comment (evelynzhaojie) [inline docs/zh/administration/Configuration.md:630]: `SHOW PROC '/transactions'` 这个需要指定到 db id 才能看到事务状态，如果不在某个 db 下的话。我没有测存算分离会返回什么样

File: docs/en/administration/Configuration.md [hunk 3]
```diff
@@ -674,6 +674,105 @@ ADMIN SET FRONTEND CONFIG ("key" = "value");
...
+- **Unit**: Seconds
+- **Default**: 180
+- **Description**: The minimum interval between AutoVacuum operations on the same partition.
+- **Introduced in**: v3.2.0
+
+##### lake_autovacuum_grace_period_minutes
+
+- **Unit**: Minutes
+- **Default**: 5
+- **Description**: The time range for retaining historical data versions. Historical data versions after Compactions within this time range are not automatically cleaned via AutoVacuum. You need to set this value greater than the maximum query time to avoid that the data accessed by running queries get deleted before the queries finish.
```
Comment (evelynzhaojie) [inline docs/en/administration/Configuration.md:730]: Historical data versions within this time range are not automatically cleaned via AutoVacuum after Compactions.

File: docs/zh/administration/Configuration.md [hunk 4]
```diff
@@ -607,6 +607,105 @@ ADMIN SET FRONTEND CONFIG ("key" = "value");
...
+##### lake_autovacuum_partition_naptime_seconds
+
+- 含义：对同一个表分区进行垃圾数据清理的最小间隔时间。
+- 单位：秒
+- 默认值：180
+- 引入版本：v3.2.0
+
+##### lake_autovacuum_grace_period_minutes
+
+- 含义：保留历史数据版本的时间范围。此时间范围内的历史数据版本不会被自动清理。您需要将该值设置为大于最大查询时间，以避免正在访问中的数据被删除导致查询失败。
```
Comment (evelynzhaojie) [inline docs/zh/administration/Configuration.md:661]: 最大查询时间 是哪个参数控制的？

File: docs/zh/administration/Configuration.md (hunk 2 above)
Comment (sduzh) [reply docs/zh/administration/Configuration.md:630]: 这里写错了，应该是`show proc '/compactions'`

File: docs/zh/administration/Configuration.md (hunk 4 above)
Comment (sduzh) [reply docs/zh/administration/Configuration.md:661]: 不是由参数决定的，是由用户SQL决定的，不固定

File: docs/zh/administration/Configuration.md [hunk 5]
```diff
@@ -607,6 +607,105 @@ ADMIN SET FRONTEND CONFIG ("key" = "value");
...
+>
+> 当 `lake_ingest_slowdown_threshold` 比配置项 `lake_compaction_score_selector_min_score` 小时，实际生效的阈值会是 `lake_compaction_score_selector_min_score`。
+
+##### lake_ingest_slowdown_ratio
+
+- 含义：导入限速比例。
+- 默认值：0.1
+- 引入版本：v3.2.0
+
+数据导入任务可以分为数据写入和数据提交（COMMIT）两个阶段，导入限速是通过延迟数据提交来达到限速的目的的，降速比例计算公式为：`(compaction_score - lake_ingest_slowdown_threshold) * lake_ingest_slowdown_ratio`。例如，数据写入阶段耗时为 5 分钟，`lake_ingest_slowdown_ratio` 为 0.1，Compaction Score 比 `lake_ingest_slowdown_threshold` 多 10，那么延迟提交的时间为 `5 * 10 * 0.1 = 5` 分钟，相当于写入阶段的耗时由 5 分钟增加到了 10 分钟，平均导入速度下降了一倍。
```
Comment (evelynzhaojie) [inline docs/zh/administration/Configuration.md:695]: “降速比例”计算公式为：`(compaction_score - lake_ingest_slowdown_threshold) * lake_ingest_slowdown_ratio`。-- 这里是不是有问题，降速比例 不是公式最后的 lake_ingest_slowdown_ratio 吗

Comment (sduzh) [reply docs/zh/administration/Configuration.md:695]: 这里确实不够准确，可能改成"延迟比例"更好

File: docs/en/administration/Configuration.md [hunk 6]
```diff
@@ -674,6 +674,105 @@ ADMIN SET FRONTEND CONFIG ("key" = "value");
...
+>
+> When `lake_ingest_slowdown_threshold` is less than `lake_compaction_score_selector_min_score`, the effective threshold will be `lake_compaction_score_selector_min_score`.
+
+##### lake_ingest_slowdown_ratio
+
+- **Default**: 0.1
+- **Description**: The ratio of the loading rate slowdown when Data Ingestion Slowdown is triggered.
+- **Introduced in**: v3.2.0
+
+Data loading tasks consist of two phases: data writing and data committing (COMMIT). Data Ingestion Slowdown is achieved by delaying data committing. The slowdown ratio is calculated with the following formula: `(compaction_score - lake_ingest_slowdown_threshold) * lake_ingest_slowdown_ratio`. For example, if the data writing phase takes 5 minutes, `lake_ingest_slowdown_ratio` is 0.1, and the Compaction Score is 10 higher than `lake_ingest_slowdown_threshold`, the delay in data committing time is `5 * 10 * 0.1 = 5` minutes, which is equivalent to doubling the average loading speed.
```
Comment (evelynzhaojie) [inline docs/en/administration/Configuration.md:762]: 1. The slowdown ratio is calculated --- not clear, see question in chinese    2. calculated using the following formula   3. which is equivalent to doubling the average loading speed. --- "doubling" is not correct, which means the average loading speed is halved.

**Review by EsoragotoSpirit — DISMISSED** [review]

**Review by EsoragotoSpirit — DISMISSED** [review]

File: docs/en/administration/Configuration.md [hunk 7]
```diff
@@ -674,6 +674,105 @@ ADMIN SET FRONTEND CONFIG ("key" = "value");
 - **Default**: `15 * 60 * 100`
 - **Description**: When the tablet clone tasks are being scheduled, if a tablet has not been scheduled for the specified time in this parameter, StarRocks gives it a higher priority to schedule it as soon as possible.
 
+#### Shared-data specific
+
+##### lake_compaction_score_selector_min_score
+
+- **Default**: 10.0
+- **Description**: The Compaction Score threshold that triggers Compaction operations. When the Compaction Score of a partition is greater than or equal to this value, the system performs Compaction on that partition.
+- **Introduced in**: v3.2.0
```
Comment (sduzh) [inline docs/en/administration/Configuration.md:683]:
```suggestion
- **Introduced in**: v3.1.0
```

File: docs/en/administration/Configuration.md [hunk 8]
```diff
@@ -674,6 +674,105 @@ ADMIN SET FRONTEND CONFIG ("key" = "value");
...
+
+The Compaction Score indicates whether a partition needs Compaction and is scored based on the number of files in the partition. Excessive number of files can impact query performance, so the system periodically performs Compaction to merge small files and reduce the file count. You can check the Compaction Score for a partition based on the `MaxCS` column in the result returned by running [SHOW PARTITIONS](../sql-reference/sql-statements/data-manipulation/SHOW_PARTITIONS.md).
+
+##### lake_compaction_max_tasks
+
+- **Default**: -1
+- **Description**: The maximum number of concurrent Compaction tasks allowed.
+- **Introduced in**: v3.2.0
```
Comment (sduzh) [inline docs/en/administration/Configuration.md:691]:
```suggestion
- **Introduced in**: v3.1.0
```

File: docs/en/administration/Configuration.md [hunk 9]
```diff
@@ -674,6 +674,105 @@ ADMIN SET FRONTEND CONFIG ("key" = "value");
...
+
+The system calculates the number of Compaction tasks based on the number of tablets in a partition. For example, if a partition has 10 tablets, performing one Compaction on that partition creates 10 Compaction tasks. If the number of concurrently executing Compaction tasks exceeds this threshold, the system will not create new Compaction tasks. Setting this item to `0` disables Compaction, and setting it to `-1` allows the system to automatically calculate this value based on an adaptive strategy.
+
+##### lake_compaction_history_size
+
+- **Default**: 12
+- **Description**: The number of recent successful Compaction task records to keep in the memory of the Leader FE node. You can view recent successful Compaction task records using the `SHOW PROC '/transactions'` command. Note that the Compaction history is stored in the FE process memory, and it will be lost if the FE process is restarted.
+- **Introduced in**: v3.2.0
```
Comment (sduzh) [inline docs/en/administration/Configuration.md:699]:
```suggestion
- **Introduced in**: v3.1.0
```

File: docs/en/administration/Configuration.md [hunk 10]
```diff
@@ -674,6 +674,105 @@ ADMIN SET FRONTEND CONFIG ("key" = "value");
...
+
+##### lake_compaction_fail_history_size
+
+- **Default**: 12
+- **Description**: The number of recent failed Compaction task records to keep in the memory of the Leader FE node. You can view recent failed Compaction task records using the `SHOW PROC '/transactions'` command. Note that the Compaction history is stored in the FE process memory, and it will be lost if the FE process is restarted.
+- **Introduced in**: v3.2.0
```
Comment (sduzh) [inline docs/en/administration/Configuration.md:705]:
```suggestion
- **Introduced in**: v3.1.0
```

File: docs/en/administration/Configuration.md [hunk 11]
```diff
@@ -674,6 +674,105 @@ ADMIN SET FRONTEND CONFIG ("key" = "value");
...
+
+- **Default**: 512
+- **Description**: The maximum number of threads for Version Publish tasks.
+- **Introduced in**: v3.2.0
+
+##### lake_autovacuum_parallel_partitions
+
+- **Default**: 8
+- **Description**: The maximum number of partitions that can undergo AutoVacuum simultaneously. AutoVaccum is the Garbage Collection after Compactions.
+- **Introduced in**: v3.2.0
```
Comment (sduzh) [inline docs/en/administration/Configuration.md:717]:
```suggestion
- **Introduced in**: v3.1.0
```

File: docs/en/administration/Configuration.md [hunk 12]
```diff
@@ -674,6 +674,105 @@ ADMIN SET FRONTEND CONFIG ("key" = "value");
...
+
+##### lake_autovacuum_partition_naptime_seconds
+
+- **Unit**: Seconds
+- **Default**: 180
+- **Description**: The minimum interval between AutoVacuum operations on the same partition.
+- **Introduced in**: v3.2.0
```
Comment (sduzh) [inline docs/en/administration/Configuration.md:724]:
```suggestion
- **Introduced in**: v3.1.0
```

File: docs/en/administration/Configuration.md [hunk 13]
```diff
@@ -674,6 +674,105 @@ ADMIN SET FRONTEND CONFIG ("key" = "value");
...
+
+##### lake_autovacuum_grace_period_minutes
+
+- **Unit**: Minutes
+- **Default**: 5
+- **Description**: The time range for retaining historical data versions. Historical data versions within this time range are not automatically cleaned via AutoVacuum after Compactions. You need to set this value greater than the maximum query time to avoid that the data accessed by running queries get deleted before the queries finish.
+- **Introduced in**: v3.2.0
```
Comment (sduzh) [inline docs/en/administration/Configuration.md:731]:
```suggestion
- **Introduced in**: v3.1.0
```

File: docs/en/administration/Configuration.md [hunk 14]
```diff
@@ -674,6 +674,105 @@ ADMIN SET FRONTEND CONFIG ("key" = "value");
...
+
+##### lake_autovacuum_stale_partition_threshold
+
+- **Unit**: Hours
+- **Default**: 12
+- **Description**: If a partition has no updates (loading, DELETE, or Compactions) within this time range, the system will not perform AutoVacuum on this partition.
+- **Introduced in**: v3.2.0
```
Comment (sduzh) [inline docs/en/administration/Configuration.md:738]:
```suggestion
- **Introduced in**: v3.1.0
```

File: docs/zh/administration/Configuration.md [hunk 15]
```diff
@@ -607,6 +607,105 @@ ADMIN SET FRONTEND CONFIG ("key" = "value");
...
+- 默认值： 10.0
+- 引入版本：v3.2.0
+
+Compaction Score 代表了一个表分区是否值得进行 Compaction 的评分，您可以通过 [SHOW PARTITIONS](../sql-reference/sql-statements/data-manipulation/SHOW_PARTITIONS.md) 语句返回中的 `MaxCS` 一列的值来查看某个分区的 Compaction Score。Compaction Score 和分区中的文件数量有关系。文件数量过多将影响查询性能，因此系统后台会定期执行 Compaction 操作来合并小文件，减少文件数量。
+
+##### lake_compaction_max_tasks
+
+- 含义：允许同时执行的 Compaction 任务数。
+- 默认值：-1
+- 引入版本：v3.2.0
```
Comment (sduzh) [inline docs/zh/administration/Configuration.md:624]:
```suggestion
- 引入版本：v3.1.0
```

File: docs/zh/administration/Configuration.md [hunk 16]
```diff
@@ -607,6 +607,105 @@ ADMIN SET FRONTEND CONFIG ("key" = "value");
...
+
+- 含义：在 Leader FE 节点内存中保留多少条最近成功的 Compaction 任务历史记录。您可以通过 `SHOW PROC '/compactions'` 命令查看最近成功的 Compaction 任务记录。请注意，Compaction 历史记录是保存在 FE 进程内存中的，FE 进程重启后历史记录会丢失。
+- 默认值：12
+- 引入版本：v3.2.0
+
+##### lake_compaction_fail_history_size
+
+- 含义：在 Leader FE 节点内存中保留多少条最近失败的 Compaction 任务历史记录。您可以通过 `SHOW PROC '/compactions'` 命令查看最近失败的 Compaction 任务记录。请注意，Compaction 历史记录是保存在 FE 进程内存中的，FE 进程重启后历史记录会丢失。
+- 默认值：12
+- 引入版本：v3.2.0
```
Comment (sduzh) [inline docs/zh/administration/Configuration.md:638]:
```suggestion
- 引入版本：v3.1.0
```

File: docs/zh/administration/Configuration.md [hunk 17]
```diff
@@ -607,6 +607,105 @@ ADMIN SET FRONTEND CONFIG ("key" = "value");
...
+
+- 含义：发送生效版本（Publish Version）任务的最大线程数。
+- 默认值：512
+- 引入版本：v3.2.0
+
+##### lake_autovacuum_parallel_partitions
+
+- 含义：最多可以同时对多少个表分区进行垃圾数据清理（AutoVacuum，即在 Compaction 后进行的垃圾文件回收）。
+- 默认值：8
+- 引入版本：v3.2.0
```
Comment (sduzh) [inline docs/zh/administration/Configuration.md:650]:
```suggestion
- 引入版本：v3.1.0
```

File: docs/zh/administration/Configuration.md [hunk 18]
```diff
@@ -607,6 +607,105 @@ ADMIN SET FRONTEND CONFIG ("key" = "value");
...
+
+##### lake_autovacuum_partition_naptime_seconds
+
+- 含义：对同一个表分区进行垃圾数据清理的最小间隔时间。
+- 单位：秒
+- 默认值：180
+- 引入版本：v3.2.0
```
Comment (sduzh) [inline docs/zh/administration/Configuration.md:657]:
```suggestion
- 引入版本：v3.1.0
```

File: docs/zh/administration/Configuration.md [hunk 19]
```diff
@@ -607,6 +607,105 @@ ADMIN SET FRONTEND CONFIG ("key" = "value");
...
+
+##### lake_autovacuum_grace_period_minutes
+
+- 含义：保留历史数据版本的时间范围。此时间范围内的历史数据版本不会被自动清理。您需要将该值设置为大于最大查询时间，以避免正在访问中的数据被删除导致查询失败。
+- 单位：分钟
+- 默认值：5
+- 引入版本：v3.2.0
```
Comment (sduzh) [inline docs/zh/administration/Configuration.md:664]:
```suggestion
- 引入版本：v3.1.0
```

File: docs/zh/administration/Configuration.md [hunk 20]
```diff
@@ -607,6 +607,105 @@ ADMIN SET FRONTEND CONFIG ("key" = "value");
...
+
+##### lake_autovacuum_stale_partition_threshold
+
+- 含义：如果某个表分区在该阈值范围内没有任何更新操作(导入、删除或 Compaction)，将不再触发该分区的自动垃圾数据清理操作。
+- 单位：小时
+- 默认值：12
+- 引入版本：v3.2.0
```
Comment (sduzh) [inline docs/zh/administration/Configuration.md:671]:
```suggestion
- 引入版本：v3.1.0
```

**Review by EsoragotoSpirit — DISMISSED** [review]

File: docs/zh/administration/Configuration.md (hunk 1 above)
Comment (evelynzhaojie) [reply docs/zh/administration/Configuration.md:618]: 那 show partitions 里还是要加一下存算分离这个字段的说明，不然用户跳转到 show partitions 了，但是没找到 maxcs 这个返回字段

**Review by evelynzhaojie — DISMISSED** [review]

**Review by EsoragotoSpirit — APPROVED** [review]

**Review by evelynzhaojie — APPROVED** [review]

**Comment by EsoragotoSpirit** [comment]
@mergify backport branch-3.1

**Pull request merged**